src/Bicep.LangServer/Handlers/BicepDisableLinterRuleHandler.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Bicep.Core;
+using MediatR;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    public class BicepDisableLinterRuleHandler : ExecuteTypedCommandHandlerBase<DocumentUri, string, string>
+    {
+        private const string bicepConfigResourceName = "Bicep.LanguageServer.bicepconfig.json";
+
+        private readonly string DefaultBicepConfig;
+
+        public BicepDisableLinterRuleHandler(ISerializer serializer)
+            : base(LanguageConstants.DisableLinterRuleCommandName, serializer)
+        {
+            var assembly = Assembly.GetExecutingAssembly();
+            var stream = assembly.GetManifestResourceStream(bicepConfigResourceName);
+            var streamReader = new StreamReader(stream ?? throw new ArgumentNullException("Stream is null"), Encoding.Default);
+
+            DefaultBicepConfig = streamReader.ReadToEnd();
+        }
+
+        public override async Task<Unit> Handle(DocumentUri documentUri, string code, string bicepConfigFilePath, CancellationToken cancellationToken)
+        {
+            (string updatedBicepConfigFilePath, string bicepConfigContents) = GetBicepConfigFilePathAndContents(documentUri, code, bicepConfigFilePath);
+            File.WriteAllText(updatedBicepConfigFilePath, bicepConfigContents);
```

Comment:
>WriteAllText [](http://example.com/codeflow?start=17&length=12)

If the user has unsaved changes in a text editor and uses the code action to disable the linter, will this cause their changes to be lost? Or will this trigger the VS code file save conflict dialog?

Reply:
Current behavior is as follows:
- If the user has unsaved changes in text editor and code action to disable linter is invoked, users changes will be lost. The rule level is set to off, bicepconfig.json on disk is updated and saved
- If the user then tries to update config file(already open) in vscode, file save conflict dialog shows up
![image](https://user-images.githubusercontent.com/30270536/134115095-fea72380-ac4d-4083-aaed-20bea701d193.png)

Do you have any concerns/suggestions?

Reply:
That dialog is super annoying when it appears.

Looking at the LSP spec for code actions (https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_codeAction), it seems that we can return workspace edits for VS code to apply to files in the workspace. For this to work right, the language server would have to have the most up to date picture of the config file contents. 

You'd have to bring back the text document change handler you had in a previous PR. We still shouldn't consume the contents during compilation if the file isn't saved, but we could use it to compute edits to the config file.

Reply:
What happens when the edits are made to the editor file though? If we don't save to disk, then applying the fix won't trigger the warning to go away.

Reply:
Unsaved file scenario is a but tricky. @majastrz and I discussed few options on Friday, but couldn't finalize on anything. I'm going to bring it up in today's DSL meeting. I plan to handle the scenario in a follow up PR.

Reply:
> What happens when the edits are made to the editor file though? If we don't save to disk, then applying the fix won't trigger the warning to go away.

With current implementation, the editor would keep edits. When the user saves them, the conflict dialog would appear if the user used the lightbulb to update the config file in between. The dialog can then be used to resolve the conflict to proceed with the save.